src/chart/generator/colorbase.h

```diff
@@ -2,29 +2,53 @@
 #define COLORBASE_H
 
 #include <cstdint>
+#include <variant>
 
 namespace Vizzu::Gen
 {
 
 struct ColorBase
 {
-	bool discrete{true};
-	uint32_t index{0};
-	double pos{0};
-	double lightness{0};
+	[[nodiscard]] double getLightness() const { return lightness; }
+
+	[[nodiscard]] uint32_t getIndex() const
+	{
+		return std::get<uint32_t>(indexOrPos);
+	}
+
+	[[nodiscard]] double getPos() const
+	{
+		return std::get<double>(indexOrPos);
+	}
+
+	void setLightness(double value) { lightness = value; }
+
+	void setIndex(uint32_t index) { indexOrPos = index; }
+
+	void setPos(double pos) { indexOrPos = pos; }
```

Comment:
These setters are not used. These are necessary?

Reply:
setLightness, and setPos are used. setIndex removed.